### ISSUE TITLE
Can't Disable Env

### ISSUE THREAD

**Issue by aldendaniels** [eslint/eslint#4463]
First, thanks for all the hard work on the ESLint project. ESLint is a fantastic gift to the JS community and makes coding more fun and more efficient every day.

I'm having trouble "turning off" an environment in my `.eslintrc` file.

My directory structure looks like this:
```
app/
  __TEST__/
    .eslintrc
  .eslintrc
  app.js
```

In my `__TEST__` directory I'd like to turn off the "browser" env so that browser globals are flagged, since my mocha tests are run on node.

Currently, `__TEST__/.eslintrc` is configured like this:
```json
{
  "env": {
    "mocha": true,
    "browser": false,
    "node": true
  }
}
```

Unfortunately, the `"browser": false` does not disable the top-level setting.

Your help is appreciated!

**NOTE:** Glob based configuration (#3611) will remove the need for this approach.

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by ilyavolodin** [comment]
Duplicate of #4379 This is a known issue/feature.

**Comment by aldendaniels** [comment]
Thanks! Missed that.
